src/metabase/api/permissions.clj

```diff
@@ -131,32 +148,77 @@
   "Fetch a map describing the group memberships of various users.
    This map's format is:
 
-    {<user-id> [{:membership_id <id>
-                 :group_id      <id>}]}"
+    {<user-id> [{:membership_id    <id>
+                 :group_id         <id>
+                 :is_group_manager boolean}]}"
   []
-  (api/check-superuser)
-  (group-by :user_id (db/select [PermissionsGroupMembership [:id :membership_id] :group_id :user_id])))
+  (validation/check-group-manager)
+  (group-by :user_id (db/select [PermissionsGroupMembership [:id :membership_id :is_group_manager]
+                                 :group_id :user_id :is_group_manager]
+                                (cond-> {}
+                                  (and (not api/*is-superuser?*)
+                                       api/*is-group-manager?*)
```

Comment:
What if we just have `*is-group-manager?*` return `false` if the user is a superuser?

i.e. we make the "group manager" concept specific to non-superusers